modules/perception/camera/app/cipv_camera.cc

```diff
@@ -887,9 +896,50 @@ bool Cipv::CollectDrops(const base::MotionBufferPtr &motion_buffer,
       pt(3) = 1.0f;
 
       Eigen::Vector3d transformed_pt;
-      TranformPoint(pt, (*motion_buffer)[motion_size - count - 1].motion,
-                    &transformed_pt);
-      //      obj->drops.emplace_back(transformed_pt);
+      accum_motion_buffer *= motion_buffer->at(motion_size - it - 1).motion;
+      TranformPoint(pt, accum_motion_buffer, &transformed_pt);
+      // TranformPoint(pt, (*motion_buffer)[motion_size - count - 1].motion,
+      //               &transformed_pt);
+      if (debug_level_ >= 3) {
+        AINFO << "(*motion_buffer)[" << motion_size - it - 1 << "].motion:";
+        AINFO << motion_buffer->at(motion_size - it - 1).motion(0, 0) << ", "
+              << motion_buffer->at(motion_size - it - 1).motion(0, 1) << ", "
+              << motion_buffer->at(motion_size - it - 1).motion(0, 2) << ", "
+              << motion_buffer->at(motion_size - it - 1).motion(0, 3) << ", ";
+        AINFO << motion_buffer->at(motion_size - it - 1).motion(1, 0) << ", "
+              << motion_buffer->at(motion_size - it - 1).motion(1, 1) << ", "
+              << motion_buffer->at(motion_size - it - 1).motion(1, 2) << ", "
+              << motion_buffer->at(motion_size - it - 1).motion(1, 3) << ", ";
+        AINFO << motion_buffer->at(motion_size - it - 1).motion(2, 0) << ", "
+              << motion_buffer->at(motion_size - it - 1).motion(2, 1) << ", "
+              << motion_buffer->at(motion_size - it - 1).motion(2, 2) << ", "
+              << motion_buffer->at(motion_size - it - 1).motion(2, 3) << ", ";
+        AINFO << motion_buffer->at(motion_size - it - 1).motion(3, 0) << ", "
+              << motion_buffer->at(motion_size - it - 1).motion(3, 1) << ", "
+              << motion_buffer->at(motion_size - it - 1).motion(3, 2) << ", "
+              << motion_buffer->at(motion_size - it - 1).motion(3, 3);
+        AINFO << "accum_motion_buffer[" << motion_size - it - 1 << "] =";
+        AINFO << accum_motion_buffer(0, 0) << ", "
+              << accum_motion_buffer(0, 1) << ", "
+              << accum_motion_buffer(0, 2) << ", "
+              << accum_motion_buffer(0, 3) << ", ";
+        AINFO << accum_motion_buffer(1, 0) << ", "
+              << accum_motion_buffer(1, 1) << ", "
+              << accum_motion_buffer(1, 2) << ", "
+              << accum_motion_buffer(1, 3) << ", ";
+        AINFO << accum_motion_buffer(2, 0) << ", "
+              << accum_motion_buffer(2, 1) << ", "
+              << accum_motion_buffer(2, 2) << ", "
+              << accum_motion_buffer(2, 3) << ", ";
+        AINFO << accum_motion_buffer(3, 0) << ", "
+              << accum_motion_buffer(3, 1) << ", "
+              << accum_motion_buffer(3, 2) << ", "
+              << accum_motion_buffer(3, 3);
```

Comment:
Better to add a helper function somewhere to pretty-print a Matrix4f, and then reuse it.

Reply:
<< supports print-out but it is ugly. Anyway I removed this and used original print-out.